atlas/atlas.go

```diff
@@ -267,9 +268,25 @@ func (a *Atlas) SetObservability(o observability.Interface) {
 			a.cacher = o.InstrumentedCache(a.cacher)
 		}
 	}
+
+	// Note: some jumping through hoops to not break compatibility here
+	// if there's more than one mvtProvider tegola would panic with
+	// "panic: duplicate metrics collector registration attempted"
+	var mvtProviders = make(map[string]bool)
 	for _, aMap := range a.maps {
+		if !mvtProviders[aMap.mvtProviderName] {
+			mvtProviders[aMap.mvtProviderName] = true
+		}
+	}
 
-		collectors, err := aMap.Collectors("tegola", o.CollectorConfig)
+	prefixWithProvider := len(mvtProviders) > 1
+
+	for _, aMap := range a.maps {
+		var prefix = "tegola"
+		if prefixWithProvider {
+			prefix = fmt.Sprintf("tegola_%s", aMap.mvtProviderName)
+		}
+		collectors, err := aMap.Collectors(prefix, o.CollectorConfig)
```

Comment:
After, digging deeper. I see what you mean that it would be a large refactor to get the mvt_provider name in, and I'm not sure; if it would work with how we are using labels. (Labels coming from the URL).

I don't think the suffix should be added here; instead, it should go into the Collector for the Map. There is no need to check to see if we only have one map, and only then suffix the mvt provider name. We can prefix the mvt provider name if it exists.

https://github.com/go-spatial/tegola/blob/master/atlas/map.go#L88-L94

Could be:
```go

	if m.mvtProviderName != "" {
		collect, ok := m.mvtProvider.(observability.Observer)
		if !ok {
			return nil, nil
		}
		return collect.Collectors(prefix+"_"+m.mvtProviderName, config)
	}

```

Reply:
Happy to comply here.

I was wondering however. Prometheus should be fine with reusing a metric, rendering a prefix unnecessary. 
Say you have `latency` in your application. Every provider, every route should send metrics to `latency` with additional meta information (provider, path, map, ..). Separating this into meaningful insights is Grafanas job in the end. 

Because on second thought, using a prefix comes with its on set of annoying perks for the user and tegola. 

Do you create lowercase provider name prefix or leave it?
Do you create mutate `my-provider-name` to `my_provider_name`?

Good luck discovering those in grafana when you do not know what we change them to.

Do you suggest we go ahead with the prefix and fix the underlying issue of registering duplicate names in a separate step? If you do not share those concerns, forget what I wrote. :)

Reply:
Sorry, for going dark. 
I think we should investigate how to make labels work. That would be the better way.
This really should be a label, the more I thought about it the more I'm convinced of that.
We should avoid doing stop-gap fixes; which I think altering the metric name would be.